lib/eeny-meeny/models/experiment.rb

```diff
@@ -23,9 +25,12 @@ def self.find_by_id(experiment_id)
     end
 
     def self.find_by_cookie_name(cookie_name)
-      if cookie_name =~ COOKIE_EXPERIMENT_ID_REGEX
-        find_by_id($1)
-      end
+      return nil unless EenyMeeny.config.experiments
+      return nil unless cookie_name =~ COOKIE_EXPERIMENT_ID_REGEX
+      experiment_id = $1
+      experiment_version = $2
+      experiment = EenyMeeny.config.experiments[experiment_id.to_sym]
+      new(experiment_id, **experiment) if experiment && experiment[:version].to_s == experiment_version
```

Comment:
Instead of re-implemented the `find_by_id` method then I'd suggest doing the follwoing:
```suggestion
      return unless cookie_name =~ COOKIE_EXPERIMENT_ID_REGEX

      experiment = find_by_id($1)
      return unless experiment && experiment.version.to_s == $2

      experiment
```
What do you think? :)